### PR TITLE
Feature/detect pose estimation

### PR DESCRIPTION
**Pull request by zoes-full-stack**
New code:
- Added AprilTag detection and pose estimation code
- Optimised drawing functions (it now draws the squares and tag id on the apriltags detected, as well as the 3D points on another window)
- Calls new detection and pose estimation functions in main.py
- Updated README.md
- Added typing to new code
- Followed Google's style sheet
- Added rvec and tvec as tuples
- Corrected orientation of 3D points
- Added last pose, new drawing and transformation code
- Used PEP8 to look for python styling issues such as whitespaces.
- Added calculations for predicted pose.
- Renamed variables following OpenCV's naming style
- Split class into separate sub classes for better readability.
- Added argparsing.
- Added constraint for image points when estimating pose (>=2) to avoid one tag being taken, (can do this in other ways so will test it out and put which one is best).
- Fixed try except statements.
- Optimised argparser.
- Fixed brackets.

### CODE REVIEW

File: aprilgroup_tracking/aprilgroup_pose_estimation/detect_pose.py [hunk 1]
```diff
@@ -0,0 +1,762 @@
...
+import numpy as np
+import cv2
+import apriltag
+from pathlib import Path
+import datetime
+import time
+from typing import List, Dict, Tuple
+
+
+class DetectAndGetPose:
```
Comment (RajivJGM) [inline aprilgroup_tracking/aprilgroup_pose_estimation/detect_pose.py]: It's usually preferable to use nouns for classes. How about "PoseDetector"?

File: aprilgroup_tracking/main.py [hunk 2]
```diff
@@ -1,37 +1,65 @@
...
+
+
+def obtain_argparsers():
+    # Create the parser
+    parser = argparse.ArgumentParser(description="Parser used for easy testing.")
+    # Add arguments
+
+    # Arg Parsers to enhance APE by using the predicted pose,
+    # if --no-enhanceape, solvePnP() is used with no extrinsic guess.
+    enhanceape_group = parser.add_mutually_exclusive_group(required=True)
```
Comment (RajivJGM) [inline aprilgroup_tracking/main.py]: You're already setting enhanceape to True by default, so we just need a flag to toggle it off.
This should be enough:
`parser.add_argument('--disable-enhanced-ape', dest='enhancedape', action='store_false', default=True)`

**Review by RajivJGM — APPROVED** [review]
Looks good! Thanks for updating the exceptions to be more specific.

File: aprilgroup_tracking/aprilgroup_pose_estimation/detect_pose.py (hunk 1 above)
Comment (zoes-full-stack) [reply aprilgroup_tracking/aprilgroup_pose_estimation/detect_pose.py]: Done! :D

File: aprilgroup_tracking/main.py (hunk 2 above)
Comment (zoes-full-stack) [reply aprilgroup_tracking/main.py]: Done! :D

**Pull request merged**